### DISCUSSION TITLE
Migration

### DISCUSSION THREAD

**Discussion by JosepAjith** [vitusortner/floor#592] [Q&A] [answered]
How migrate floor Database

**Comment by vitusortner** [comment, marked as answer]
You can find instructions here: https://floor.codes/migrations/

**Reply by neeluagrawal04** [reply]
can't open the link

**Reply by Amir-P** [reply]
Check this out: https://vitusortner.github.io/floor/migrations/ @neeluagrawal04

**Reply by zmm2tysu** [reply]
@Amir-P can't open either

**Reply by ThomasMiddel** [reply]
@zmm2tysu You can find the updated link here: https://pinchbv.github.io/floor/migrations/

**Comment by Soliardi-Chyngyz** [comment]
Anyone could help me with primery keys?

I added: skiPass, date
https://github.com/entity(primaryKeys: ['entityType', 'id', 'skiPass', 'date'])

but this doesn't work and show error
final migration1to2 = Migration(1, 2, (database) {
return database
.execute('ALTER TABLE TariffEntity DROP PRIMARY KEY, ADD PRIMARY KEY(skiPass, date)');
});

tried 'ALTER TABLE TariffEntity DROP PRIMARY KEY, ADD PRIMARY KEY(skiPass, date)' nothing

**Reply by dkaera** [reply]
Hey, SQLite doesn't support replacing the primary key in this way.
So, you need to follow next steps:

```dart
// Create copy of table with a new PRIMARY KEY
await database. execute ( 'CREATE TABLE TariffEntityCopy(id INTEGER, skiPass TEXT, date TEXT, PRIMARY KEY (skiPass, date));
// Copy all data from the original to the copy
await database. execute ('INSERT INTO my_table_copy (id, skiPass, date) SELECT id, skiPass, date FROM TariffEntity');
// Copy drop the original table
await database.execute ('DROP TABLE TariffEntity');
// Rename copy of the table to original
await database. execute ( 'ALTER TABLE TariffEntityCopy RENAME TO TariffEntity');
```
